component/src/main/java/org/wso2/extension/siddhi/execution/geo/GeoCoordinateStreamFunctionProcessor.java

```diff
@@ -0,0 +1,216 @@
+                        defaultValue = "N/A",
+                        possibleParameters = "N/A"
+                )
+        },
+        examples = @Example(
+                description = "This will return the longitude and latitude of the given IPV4 or IPV6 address. " +
+                        "So the results for the geocoordinate(95.31.18.119) are 55.7522, 37.6156",
+                syntax = "define stream IpStream(ip string); " +
+                        "from IpStream#geo:geocoordinate(ip) " +
+                        "select latitude, longitude " +
+                        "insert into outputStream;")
+)
+
+public class GeoCoordinateStreamFunctionProcessor extends StreamFunctionProcessor {
+    private static GeoCoordinateResolver geoCoordinateResolverImpl;
+    private static final String DEFAULT_GEOCOORDINATE_RESOLVER_CLASSNAME =
+            "org.wso2.extension.siddhi.execution.geo.internal.impl.APIBasedGeoCoordinateResolver";
+    private static AtomicBoolean isExtensionConfigInitialized = new AtomicBoolean(false);
+
+    /**
+     * The process method used when more than one function parameters are provided
+     *
+     * @param data the data values for the function parameters
+     * @return the data for additional output attributes introduced by the function
+     */
+    @Override
+    protected Object[] process(Object[] data) {
+        return new Object[0];
+    }
+
+    /**
+     * The process method used when zero or one function parameter is provided
+     *
+     * @param data null if the function parameter count is zero or runtime data value of the function parameter
+     * @return the data for additional output attribute introduced by the function
+     */
+    @Override
+    protected Object[] process(Object data) {
+        String ip = data.toString();
+        GeoCoordinate geoCoordinate = geoCoordinateResolverImpl.getGeoCoordinateInfo(ip);
+        return new Object[]{geoCoordinate.getLatitude(), geoCoordinate.getLongitude()};
+    }
+
+    /**
+     * The initialization method for {@link StreamFunctionProcessor},
+     * which will be called before other methods and validate
+     * the all configuration and getting the initial values.
+     *
+     * @param inputDefinition              the incoming stream definition
+     * @param attributeExpressionExecutors the executors of each function parameters
+     * @param configReader                 this hold the {@link StreamFunctionProcessor} extensions
+     *                                     configuration reader.
+     * @param siddhiAppContext             the context of the siddhi app
+     * @return the output's additional attributes list introduced by the function
+     */
+    @Override
+    protected List<Attribute> init(AbstractDefinition inputDefinition,
+                                   ExpressionExecutor[] attributeExpressionExecutors, ConfigReader configReader,
+                                   SiddhiAppContext siddhiAppContext) {
```

Comment:
Should we not validate the length of attributeExpressionExecutors here (that it should be one)?

Reply:
Fixed with commit : https://github.com/wso2-extensions/siddhi-execution-geo/pull/50/commits/2789a52effed321109fb15b163c8dbc908df4059